pkg/discovery/sqlite_backend.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package discovery
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	// Import the sqlite3 driver
+	_ "github.com/mattn/go-sqlite3"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/catalog"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/common"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/distribution"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
+)
+
+const (
+	// sqliteDBFileName is the name of the DB file that is stored in
+	// the OCI image describing the content of the Central Repository
+	sqliteDBFileName = "central.db"
```

Comment:
Thanks for updating the Discovery name and factoring out the db access bit.

nit: (If we want we can address in a followup), I see there is still a fair bit of references to central repository, 
in comments, types and variables such as this. 
I think presenting the implementation more generally does not preclude using it specifically for the Central Repo, but painting this as meant for central repo then finding we might have to use it in other scenarios will lead to more awkwardness. 

e.g. the db file here could be named something like plugins.db or plugin_inventory.db and still make sense.

Reply:
Agreed.  I will clean this up.  BTW I much prefer your use of the term "inventory" instead of "content" that I've been using everywhere.  Will update that too 👍

Reply:
I've cleaned all that up and removed all references to a "central repo", except in the commit text and the feature-flag name.